activerecord/test/cases/associations_test.rb

```diff
@@ -329,6 +329,53 @@ def test_requires_symbol_argument
       end
     end
   end
+
+  # We are creating a special class here, to avoid mangling any other models with a `class_eval`
+  class SpecialPirate1 < ActiveRecord::Base
+    self.table_name = 'pirates'
+  end
+
+  class SpecialPirate2 < SpecialPirate1
```

Comment:
This subclass needs an association of some kind in order to reproduce the issue properly: https://gist.github.com/matthewd/8b88b7356d4076fe6a10#file-gistfile1-rb-L20.